src/body_operations.cpp

```diff
@@ -39,29 +39,40 @@
 #include <console_bridge/console.h>
 #include <Eigen/Geometry>
 
+bodies::Body* bodies::createEmptyBodyFromShapeType(const shapes::ShapeType& shapeType)
+{
+  Body* body = nullptr;
+
+  switch (shapeType)
+  {
+    case shapes::BOX:
+      body = new bodies::Box();
+      break;
+    case shapes::SPHERE:
+      body = new bodies::Sphere();
+      break;
+    case shapes::CYLINDER:
+      body = new bodies::Cylinder();
+      break;
+    case shapes::MESH:
+      body = new bodies::ConvexMesh();
+      break;
+    default:
+      CONSOLE_BRIDGE_logError("Creating body from shape: Unknown shape type %d", (int)shapeType);
+      break;
+  }
+  return body;
+}
+
 bodies::Body* bodies::createBodyFromShape(const shapes::Shape* shape)
 {
   Body* body = nullptr;
 
   if (shape)
-    switch (shape->type)
-    {
-      case shapes::BOX:
-        body = new bodies::Box(shape);
-        break;
-      case shapes::SPHERE:
-        body = new bodies::Sphere(shape);
-        break;
-      case shapes::CYLINDER:
-        body = new bodies::Cylinder(shape);
-        break;
-      case shapes::MESH:
-        body = new bodies::ConvexMesh(shape);
-        break;
-      default:
-        CONSOLE_BRIDGE_logError("Creating body from shape: Unknown shape type %d", (int)shape->type);
-        break;
-    }
+  {
+    body = createEmptyBodyFromShapeType(shape->type);
+    body->setDimensions(shape);
```

Comment:
Doesn't that change the semantics of `createBodyFromShape()`?
Before it wasn't calling `setDimensions()`.

Reply:
This doesn't change any semantics. The `Shape*` constructor calls `setDimensions()` internally. E.g. here: https://github.com/ros-planning/geometric_shapes/blob/c17a1de10c4befd9e220b9c79afa5df0cda85154/include/geometric_shapes/bodies.h#L454 .

Modifying this function is not needed. It doesn't bring any speedup. It just shortens the code.